member_map.go

```diff
@@ -198,10 +190,48 @@ func override(newMem Member, existingMem Member) Member {
 	return existingMem
 }
 
-// Update member and update waitingMembers
-// todo
-func (m *MemberMap) UpdateMember(member Member) error {
-	return nil
+// Process Alive message
+// Return 2 parameter bool, error bool means Changed in MemberList
+// 1. If aliveMessage Id is empty return false and ErrEmptyMemberID
+// 2. if aliveMessage Id is not in MemberList then Create Member and Add MemberList
+// 3. if aliveMessage Id is in MemberList and existingMember's Incarnation is bigger than AliveMessage's Incarnation Than return false, ErrIncarnation
+// 4. if aliveMessage Id is in MemberList and AliveMessage's Incarnation is bigger than existingMember's Incarnation Than Update Member and Return true, nil
+func (m *MemberMap) Alive(aliveMessage AliveMessage) (bool, error) {
+
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	// Check whether Member Id empty
+	if aliveMessage.ID == "" {
+		return false, ErrEmptyMemberID
+	}
+	// Check whether it is already exist
+	existingMem, ok := m.members[MemberID{aliveMessage.ID}]
+
+	// if Member is not exist in MemberList
+	if !ok {
+		m.members[MemberID{aliveMessage.ID}] = updateMember(aliveMessage.MemberMessage, Alive)
+		return true, nil
+	}
+	// Check incarnation
+	if aliveMessage.Incarnation < existingMem.Incarnation {
+		return false, ErrIncarnation
+	}
```

Comment:
```
1. {Alive Ml, inc=i} overrides
     - {Suspect Ml, inc=j}, i>j
     - {Alive Ml, inc=j}, i>j
```

According to the override rule above, if i == j, we do not override it, so we need to keep the existing status.